pkgs/pubspec_parse/lib/src/dependency.dart

```diff
@@ -46,10 +46,8 @@ Dependency? _fromJson(Object? data, String name) {
   }
 
   if (data is Map) {
-    final matchedKeys = data.keys
-        .cast<String>()
-        .where((key) => key != 'version')
-        .toList();
+    final matchedKeys =
+        data.keys.cast<String>().where((key) => key != 'version').toList();
 
     if (data.isEmpty || (matchedKeys.isEmpty && data.containsKey('version'))) {
```

Comment:
The `data.containsKey('version')` here is unnecessary, `matchedKeys` contains every key other than `"version"`.

This can be just `if (matchedKeys.isEmpty)`.

(Not part of this CL, just cannot unsee.)